samples/algorithms/repeat-until-success/RepeatUntilSuccess.qs

```diff
@@ -0,0 +1,110 @@
+namespace Microsoft.Quantum.Samples.RepeatUntilSuccess {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Preparation;
+
+    /// Example of a Repeat-until-success circuit
+    /// The circuit is executed on a "target" qubit using an "ancilla" and 
```

Comment:
In general, I'd suggest avoiding calling Q# programs (especially those with measurement adaptivity) circuits, since that can confuse with lower-level non-adaptive constructs.

Reply:
I'll try to be more explicit about which part is the program and which part is the circuit.